tools/wpt/revlist.py

```diff
@@ -0,0 +1,299 @@
+    '''
+    git = get_git_cmd(wpt_root)
+    command = [
+        "--no-walk",
+        "--skip=%s" % skip,
+        "--max-count=%s" % max_count,
+        "--min-age=%s" % min_age,
+        "--format=epoch:%ct:commit:%H:%D",
+        commit
+    ]
+    result = {}
+    for output_line in git("rev-list", *command).split("\n")[1::2]:
+        # print output_line
+        output = output_line.split(":")
+        if len(output) > 5 and output[5].find(TAGGED_REVISIONS) > -1:
+            revision = {}
+            revision["epoch"] = int(output[1])
+            revision["commit"] = output[3].strip()
+            revision["tag"] = output[5].strip()
+            result[int(output[1])] = revision
+    return result
+
+
+def get_newer_tagged_revision(tagged_revisions, min_age):
+    # type: (Dict, int) -> Dict
+    epoch = max(i for i in tagged_revisions.keys() if i < min_age)
+    c = tagged_revisions.keys()
+    c.sort(reverse=True)
+    return tagged_revisions[epoch]
+
+
+def list_tagged_revisons(**kwargs):
+    # type: (**Any) -> List[Text]
+    head_revision = COMMAND_ARGS["head_revision"]["parser"](**kwargs)
+    logger.debug("list_tagged_revisons: %s" % kwargs)
+    epoch_step = COMMAND_ARGS["epoch_step"]["parser"](**kwargs)
+    epoch_threshold = COMMAND_ARGS["epoch_threshold"]["parser"](**kwargs)
+    epoch_until = COMMAND_ARGS["epoch_until"]["parser"](**kwargs)
+    num_revisions = COMMAND_ARGS["num_revisions"]["parser"](**kwargs)
+    verbose = COMMAND_ARGS["verbose"]["parser"](**kwargs)
+
+    if epoch_until > DEFAULT_EPOCH_UNTIL - epoch_threshold:
+        epoch_until = DEFAULT_EPOCH_UNTIL - epoch_threshold
+        logger.debug("epoch-until adjusted to %s using the safety epoch-threshold value (%s)" % (epoch_until, epoch_threshold))  # noqa E501
+    iterator = 0
+    result = []
+    previous_revision = None
+
+    # Iterates the rev-list command in bunchs of "rev_list_step"
+    # items skipping the previous "rev_list_offset" and it is incremented in
+    # "rev_list_step" each time no more valid tagged revisions are found in
+    # the current set of revisions.
+    # The loop ends once we reached the required number of revisions to return
+    # or the are no more tagged revisions returned found by the rev-list
+    # iterator.
+    # Because it is posible to find the same revision as candidate for two
+    # consecutives "epoch_step" (case I in the example) the loop checks if
+    # previous revision added is equal to the current one and skip this on
+    # positive case.
+    #
+    #   Fri   Sat   Sun   Mon   Tue   Wed   Thu   Fri   Sat
+    #    |     |     |     |     |     |     |     |     |
+    # -A---B-C---DEF---G---H--I-----------J-----K-L----M--N--
+    #                                                       ^
+    #                                                      now
+    rev_list_offset = 0
+    rev_list_step = REV_LIST_COUNT
+    tagged_revisions = get_tagged_revisions(head_revision, epoch_until,
```

Comment:
Can you ensure/document that `get_tagged_revisions` returns a list sorted by date and then simplify this to a single iteration through that list, or pairs of entries on that list perhaps?

I think the basic structure should be:
* Find the current time minus the grace period of 10 minutes.
* Filter out/ignore any information from past that point in time.

Then one of these approaches:
* Step through pairs of revisions and check if an epoch boundary is between them, and in those cases printing the revision before the epoch boundary. This will be O(n).
* More similar to what you have here: step through each epoch boundary (e.g. midnights) and find the latest revision before each step. Print if it hasn't been printed before. This could be O(n) if you keep track of where in the list of revisions you are, or at least O(n log n) if you use https://docs.python.org/2/library/bisect.html.